routing/turns_tts_text_i18n.cpp

```diff
@@ -0,0 +1,169 @@
+    // therefore the string is not exclusively uppercase/numeric
+    else if (myUniStr[i] == lowerStr[i] && !std::isdigit(myUniStr[i])) {
+      allUppercaseNum = false;
+      break;
+    }
+  }
+  return allUppercaseNum;
+}
+
+uint8_t CategorizeHungarianAcronymsAndNumbers(std::string const & hungarianString)
+{
+  constexpr std::array<std::string_view, 14> backNames = {
+      "A", // a
+      "Á", // á
+      "H", // há
+      "I", // i
+      "Í", // í
+      "K", // ká
+      "O", // o
+      "Ó", // ó
+      "U", // u
+      "Ű", // ú
+      "0", // nulla or zéró
+      "3", // három
+      "6", // hat
+      "8", // nyolc
+  };
+
+  constexpr std::array<std::string_view, 31> frontNames = {
+      // all other letters besides H and K
+      "B", "C", "D", "E", "É", "F", "G", "J", "L", "M", "N", "Ö", "Ő", "P", "Q", "R", "S", "T", "Ú", "Ü", "V", "W", "X", "Y", "Z",
+      "1", // egy
+      "2", // kettő
+      "4", // négy
+      "5", // öt
+      "7", // hét
+      "9", // kilenc
+  };
+
+  constexpr std::array<std::string_view, 5> specialCaseFront = {
+      "10", // tíz special case front
+      "40", // negyven front
+      "50", // ötven front
+      "70", // hetven front
+      "90", // kilencven front
+  };
+
+  constexpr std::array<std::string_view, 4> specialCaseBack = {
+      "20", // húsz back
+      "30", // harminc back
+      "60", // hatvan back
+      "80", // nyolcvan back
+  }; //TODO: consider regex
+
+  //'100', // száz back
+
+
+  for (size_t i = hungarianString.length(); i >= 0; i--) {
+    std::string hungarianSub = hungarianString.substr(0, i);
+    for (auto myCase : specialCaseFront)
+      if (strings::EndsWith(hungarianSub,myCase))
+        return 1;
+    for (auto myCase : specialCaseBack)
+      if (strings::EndsWith(hungarianSub,myCase))
+        return 2;
+    if (strings::EndsWith(hungarianSub,"100"))
+      return 2;
+
+    for (auto myCase : frontNames)
+      if (strings::EndsWith(hungarianSub,myCase))
+        return 1;
+    for (auto myCase : backNames)
+      if (strings::EndsWith(hungarianSub,myCase))
+        return 2;
+    if (strings::EndsWith(hungarianSub," "))
+      return 2;
+  }
+
+  LOG(LWARNING, ("Unable to find Hungarian front/back for", hungarianString));
+  return 2;
+}
+
+uint8_t CategorizeHungarianLastWordVowels(std::string const & hungarianString)
+{
+  constexpr std::array<char32_t, 6> front = {U'e',U'é',U'ö',U'ő',U'ü',U'ű'};
+  constexpr std::array<char32_t, 6> back = {U'a',U'á',U'o',U'ó',U'u',U'ú'};
+  constexpr std::array<char32_t, 2> indeterminate = {U'i',U'í'};
+
+  strings::UniString myUniStr = strings::MakeUniString(strings::MakeLowerCase(hungarianString));
+
+  // scan for acronyms and numbers first (i.e. characters spoken differently than words)
+  // if the last word is an acronym/number like M5, check those instead
+  if (EndsInAcronymOrNum(myUniStr)) {
+    return CategorizeHungarianAcronymsAndNumbers(hungarianString);
+  }
```

Comment:
@biodranik do you have a linter for this kind of stuff? If we delay PRs for all these nits I'll have an average contribution rate of 1 PR per 3 years.

Reply:
Commit your changes first. Then use `git clang-format <commit_hash>`

There could be some issues, as we didn't finish polishing it here https://github.com/organicmaps/organicmaps/pull/4755

Reply:
@biodranik ok this works with the commit *prior to the one you want to edit* and also does all commits since (so it mangles locale json files etc) but otherwise thanks this will be useful to reduce nits